include/boost/gil/image_processing/hessian.hpp

```diff
@@ -15,19 +16,18 @@ namespace boost { namespace gil {
 /// ddxx means taking two derivatives (gradients) in horizontal direction.
 /// Weights change perception of surroinding pixels.
 /// Additional filtering is strongly advised.
-template <typename GradientView, typename Weights, typename OutputView>
+template <typename GradientView, typename T, typename Allocator, typename OutputView>
 inline void compute_hessian_responses(
     GradientView ddxx,
     GradientView dxdy,
     GradientView ddyy,
-    Weights weights,
+    const kernel_2d<T, Allocator>& weights,
```

Comment:
Minor stylistic remark: since in C++ this reads from right to left as _"`weights` is a reference **to**  const `kernel_2d` object"_, as we do it for pointers, this should be the [East Const](http://slashslash.info/eastconst/) way: `kernel_2d<T, Allocator> const& weights`

You may want to join the revolution! 😉 

![image](https://user-images.githubusercontent.com/80741/67425761-f396d400-f5d8-11e9-81e9-20d7910f7c52.png)